ui/app/components/send_/send.utils.js

```diff
@@ -178,6 +165,42 @@ function doesAmountErrorRequireUpdate ({
   return amountErrorRequiresUpdate
 }
 
+async function estimateGas ({ selectedAddress, selectedToken, data, blockGasLimit, to }) {
+  const ethQuery = new EthQuery(global.ethereumProvider)
+  const { symbol } = selectedToken || {}
+  const estimatedGasParams = { from: selectedAddress }
+
+  if (symbol) {
+    Object.assign(estimatedGasParams, { value: '0x0' })
+  }
+
+  if (data) {
+    Object.assign(estimatedGasParams, { data })
+  }
+  // if recipient has no code, gas is 21k max:
+  const hasRecipient = Boolean(to)
+  let code
+  if (hasRecipient) code = await ethQuery.getCode(to)
+
+  if (hasRecipient && (!code || code === '0x')) {
+    return SIMPLE_GAS_COST
+  }
```

Comment:
Can lines 182-187 be a single block?

Reply:
Yeah I meant having the nested `if` blocks:

```
if (hasRecipient) {
  const code = await global.eth.getCode(to)
  if (!code || code === '0x') {
    return SIMPLE_GAS_COST
  }
}
```

Is this the same? It was just weird to me to check `hasRecipient` twice, plus we'd get rid of `let code`.